### PR TITLE
Add cargo-lambda

### PR DESCRIPTION
**Pull request by jayvdb**
Close https://github.com/taiki-e/install-action/issues/567

### CODE REVIEW

File: TOOLS.md [hunk 1]
```diff
@@ -4,8 +4,6 @@ This is a list of tools that are installed from manifests managed in this action
 
 If a tool not included in the list below is specified, this action uses [cargo-binstall] as a fallback.
 
-See the [Supported tools section in README.md](README.md#supported-tools) for how to ensure that fallback is not used.
```
[the remark this answers is not in the document]
Comment (taiki-e) [reply TOOLS.md:7]: This should be fixed by rebasing and regenerating TOOLS.md.

[the remark this answers is not in the document]
Comment (jayvdb) [reply TOOLS.md]: Fixed

File: tools/codegen/base/cargo-lambda.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  "asset_name": "${package}-v${version}.${rust_target}.tar.gz",
+  "version_range": ">= 1.3.0",
+  "platform": {
+    "x86_64_linux_musl": {},
+    "x86_64_macos": {},
+    "x86_64_windows": {
+      "asset_name": "${package}-v${version}.windows-x64.zip",
+      "bin": "${package}${exe}"
+    },
+    "aarch64_macos": {}
```
Comment (taiki-e) [inline tools/codegen/base/cargo-lambda.json:14]: It seems they distribute the aarch64_musl binary:
https://github.com/cargo-lambda/cargo-lambda/releases/tag/v1.3.0
> cargo-lambda-v1.3.0.aarch64-unknown-linux-musl.tar.gz

Comment (taiki-e) [inline tools/codegen/base/cargo-lambda.json:14]:
```suggestion
    "aarch64_linux_musl": {},
    "aarch64_macos": {}
```

**Review by taiki-e — APPROVED** [review]
Thanks!

**Pull request merged**